app/src/main/java/fr/free/nrw/commons/upload/SingleUploadFragment.java

```diff
@@ -188,8 +188,10 @@ public void onDestroyView() {
         String desc = titleDesc.getString("Desc", "");
         Timber.d("Title: %s, Desc: %s", title, desc);
 
-        titleEdit.setText(title);
-        descEdit.setText(desc);
+        if (title.trim().length() > 0 && desc.trim().length() > 0) {
+            descEdit.setText(desc);
+            titleEdit.setText(title);
+        }
```

Comment:
This indeed prevents the removal of the texts right after the user clicks the button. However, isn't it better to hide the button? Alternatively, you could perhaps show a Toast saying "no previous title/description is stored".